cli/src/pcluster/api/controllers/common.py

```diff
@@ -177,3 +178,12 @@ def get_validator_suppressors(suppress_validators: Optional[List[str]]) -> Set[V
         validator_suppressors.add(TypeMatchValidatorsSuppressor(validator_types_to_suppress))
 
     return validator_suppressors
+
+
+def assert_supported_operation(operation: Operation, region: str):
+    """Raise a BadRequestException if the operation is not supported in the region."""
+    validation_failures = OperationRegionValidator().execute(operation=operation, region=region)
```

Comment:
I think what you implemented is not a Validator.

The Validators are semantic validators that checks configuration parameters (and that can be "suppressed", if needed). They are automatically called by internal methods, when checking the cluster config.

Here we're using Validators logic to do instead a command check. We're forcing the execution of the `execute` method and then we're parsing the output to retrieve the `message` of the first item in the validation failures.

I think in this case it's more clear to have the `assert_supported_operation` that simply executes the content of the `_is_operation_supported` and then raises a BadRequestException, without putting the `Validator` class in the middle.

Reply:
I agree, thanks